src/wazuh.py

```diff
@@ -112,7 +115,11 @@ def configure_git(
         group=WAZUH_GROUP,
         permissions=0o600,
     )
-    process = container.exec(["git", "clone", custom_config_repository, REPOSITORY_PATH])
+    command = ["git", "clone"]
+    if branch:
+        command = command + ["--branch", branch]
+    command = command + [url, REPOSITORY_PATH]
+    process = container.exec(command)
```

Comment:
Out of curiosity, would this ever run a second time? (e.g. on update charm?) I've had cases where git clone the second time would fail so I added a check to see if the git repo already exists and delete it before I proceed with the setup commands.

Reply:
The intent is not to run it in those cases